ls1intum/Artemis
# `Programming exercises`: Fix display of long feedback details and submission count in exams

### Checklist
#### General


#### Server

### Motivation and Context

The `FeedbackDTO` did not contain the ID of the feedback. This causes that in the feedback detail view a potential long feedback can no longer be fetched, since the ID of the feedback it belongs to is required to construct the URL and know on the server which feedback should be fetched.
Currently it tries to fetch the long feedback from an URL like
https://ARTEMIS/api/results/123456789/feedbacks/undefined/long-feedback
which fails since `undefined` is not a valid id.

### Description
Adds the ID of the feedback to the DTO.
Then the feedback can be fetched from the server again.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming exercise

1. Add a test case in the exercise test repository that contains a long feedback, e.g. by placing a `fail("abc".repeat(10_000))` method call inside.
2. Instructor view of the exercise: ‘Edit in Editor’ to open the online editor, select the solution repository.
3. Submit this solution, the ‘Building and Testing’ spinner is started.
4. When the result from the CI system is returned, open the feedback details and then the ‘See more’ for the test case with the long feedback.
5. The additional text should be loaded without error.


### Review Progress






#### Code Review
#### Manual Tests
